File: WebApplication/src/modules/student/dtos/person-lot.dtos/base.dtos/base-pe.dto.ts
import { NomenclatureDto } from "src/infrastructure/models/nomenclature.dto";
import { PreviousEducationType } from "src/modules/student/enums/previous-education-type.enum";
import { PreviousHighSchoolEducationType } from "src/modules/student/enums/previous-high-school-education-type.enum";

export class BasePeDto {
  peType: PreviousEducationType;
  peHighSchoolType?: PreviousHighSchoolEducationType;

  peDiplomaNumber: string;
  peDiplomaDate: Date;
  peResearchArea: NomenclatureDto;

  peEducationalQualification: NomenclatureDto;
  peAcquiredForeignEducationalQualification: NomenclatureDto;

  peInstitution: NomenclatureDto;
  peSubordinate: NomenclatureDto;
  peSpeciality: NomenclatureDto;

  peInstitutionName: string;
  peSubordinateName: string;

  peSpecialityName: string;

  peCountry: NomenclatureDto;
  peRecognizedSpeciality: string;
  peAcquiredSpeciality: string;

  peRecognitionNumber: string;
  peRecognitionDate: Date;
}

File: StudentCard.Application/Students/Dtos/PersonLotDtos/BaseDtos/BasePeDto.cs
using StudentCard.Data.Rdpzsd.Enums;
using StudentCard.Infrastructure.Helpers.Dtos;
using System;

namespace StudentCard.Application.Students.Dtos.PersonLotDtos.BaseDtos
{
    public class BasePeDto
    {
        public PreviousEducationType PeType { get; set; }
        public PreviousHighSchoolEducationType? PeHighSchoolType { get; set; }

        public string PeDiplomaNumber { get; set; }
        public DateTime? PeDiplomaDate { get; set; }
        public ResearchAreaDto PeResearchArea { get; set; }

        //RecognizedEQ
        public NomenclatureDto PeEducationalQualification { get; set; }
        public NomenclatureDto PeAcquiredForeignEducationalQualification { get; set; }

        public NomenclatureDto PeInstitution { get; set; }
        public NomenclatureDto PeSubordinate { get; set; }
        public NomenclatureDto PeSpeciality { get; set; }

        // Abroad, ClosedInstitution
        public string PeInstitutionName { get; set; }
        public string PeSubordinateName { get; set; }

        // ClosedInstitution
        public string PeSpecialityName { get; set; }

        // Abroad
        public NomenclatureDto PeCountry { get; set; }
        public string PeRecognizedSpeciality { get; set; }
        public string PeAcquiredSpeciality { get; set; }

        public string PeRecognitionNumber { get; set; }
        public DateTime? PeRecognitionDate { get; set; }
    }
}
